### ISSUE TITLE
[Task Module]: Android - Failed to load web view

### ISSUE THREAD

**Issue by garrytrinder** [OfficeDev/microsoft-teams-library-js#1582]
Related to https://github.com/garrytrinder/msteams-dalle-2-image-generator-bot/issues/38

Repro steps

- Implement Adaptive Card containing `Image` element that uses `$data` looping, each image given a `selectAction` that sends a `task/fetch` activity and data payload. (ex: 
- Implement `handleTeamsTaskModuleFetch` method that returns a `TaskModuleResponse` with URL to HTML page that passes data through query string parameters (ex: https://github.com/garrytrinder/msteams-dalle-2-image-generator-bot/blob/385d0962eb64ba5e472107734cbec91389f8a970/bot/src/internal/activityHandler.ts#L15-L29)
- Implement HTML page that takes the values from query string and updates an HTML image tag to render an image (ex: https://github.com/garrytrinder/msteams-dalle-2-image-generator-bot/blob/385d0962eb64ba5e472107734cbec91389f8a970/bot/src/public/image.html#L17-L28)

Works fine on web client and iOS, but not Android.

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - Thanks for reporting your issue. We will test this and update you soon.

**Comment by garrytrinder** [comment]
Thanks for the reply @Nivedipa-MSFT.

> Console error which you are getting in android.

I'm not sure how to obtain this from within the Teams app.

I am using a Virtual Device in Android Studio to test the app.

> Valid domains list in the manifest.

Below is the `validDomains` array from the manifest.

```json
"validDomains": [
  "9a0a-86-135-49-130.ngrok.io",
  "oaidalleapiprodscus.blob.core.windows.net"
]
```

> URL which the task module is hitting.

Below is the full `TaskModuleResponse` containing the URL.
 
```json
{
  "task": {
    "type": "continue",
    "value": {
      "url": "https://9a0a-86-135-49-130.ngrok.io/image.html?imgUrl=https%3A%2F%2Foaidalleapiprodscus.blob.core.windows.net%2Fprivate%2Forg-9kI5Jg63z5MfzS16zTNDWXBP%2Fuser-cIzmxziBZ7iaJjbV0zC5YO89%2Fimg-U0xnPFAH1bFZmJKLqM5Lwbcw.png%3Fst%3D2023-02-15T11%253A06%253A22Z%26se%3D2023-02-15T13%253A06%253A22Z%26sp%3Dr%26sv%3D2021-08-06%26sr%3Db%26rscd%3Dinline%26rsct%3Dimage%2Fpng%26skoid%3D6aaadede-4fb3-4698-a8f6-684d7786b067%26sktid%3Da48cca56-e6da-484e-a814-9c849652bcb3%26skt%3D2023-02-15T02%253A29%253A38Z%26ske%3D2023-02-16T02%253A29%253A38Z%26sks%3Db%26skv%3D2021-08-06%26sig%3DExP2qf4oikhMdtAn4VesTP7IjMnpVwstJjpntqXueu0%253D&prompt=%22a%20sea%20otter%20with%20a%20pearl%20earring%22%20by%20Johannes%20Vermeer",
      "height": 512,
      "width": 512,
      "title": "\"a sea otter with a pearl earring\" by Johannes Vermeer"
    }
  }
}
```

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - Using DevTools you can get the console logs on Android device.
Access DevTools from an Android device.
https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/developer-tools#access-devtools-from-an-android-device

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - Did you get chance to check console logs on Android device?

**Comment by garrytrinder** [comment]
Unfortunately the DevTools don't provide me with a way to debug the bot running in Teams on the Android device. See the output below. The device is detected but it provides me with no options to debug.

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/11563347/220467581-c0ef987e-ee2b-437c-9119-6cb54f2086d3.png">

I have included `localhost:3798`(the local webserver) in the Target discovery settings. Am I missing some other configuration to debug the bot?

I have been able to debug tabs running in Teams on Android previously so I know this works on my current setup.

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - Have you set up your Android device for [remote debugging](https://developers.google.com/web/tools/chrome-devtools/remote-debugging/)?

**Comment by garrytrinder** [comment]
Yes, when I start the device a prompt is opened asking for consent to allow remote debugging, which I allow. I will double check the settings.

**Comment by garrytrinder** [comment]
I can confirm that I have the following configured

- Virtual Device (Pixel 4) is in Developer Mode.
- USB and Wireless Debug options are enabled on Device in Developer Options.
- Remote debug prompts set to always allow.
- Developer Preview is enabled on Teams app (1416/1.0.0.2022445001/0110).

The device is detected in Edge but no options shown to debug an app.

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - Could you please test this with the Chrome browser if the debug option is coming or not?

**Comment by garrytrinder** [comment]
I've tried on Google Chrome and I get the same experience, I cannot debug the app.

Worth noting that I am on macOS. Did your tests happen on Windows?

**Comment by ChetanSharma-msft** [comment]
Hello @garrytrinder - Yes, we have tested it in Windows machine.
We will test it in MacOS as well and let you know the updates.

**Comment by Nivedipa-MSFT** [comment]
@garrytrinder - We have tested this again, but it was working at our end. Could you please configure the logs to debug the application.
For configuring logs please refer below document:
https://learn.microsoft.com/en-us/microsoftteams/log-files

**Comment by garrytrinder** [comment]
@Nivedipa-MSFT thanks for the reply, let me take a look and get back to you.
